Statistics/Histograms/Histogram.cs

```diff
@@ -32,127 +38,218 @@ internal abstract class Histogram : IHistogram
+                deviation3 += deviation2 * deviation;
+
+            }
+            double skewness = SampleSize > 2 ? deviation3 / SampleSize / Math.Pow(Variance, 3 / 2) : 0;
+            return skewness;
+        }
+
+
 
         #region Functions
-        public static IHistogram Fit(IData data, int nBins = 100) => IHistogramFactory.Factory(data, nBins);
-        public static IHistogram Fit(IHistogram histogram, IData sample)
+        public void AddObservationToHistogram(IData data)
+        {   
+            Int64 quantityAdditionalBins = 0;
+
+            if (data.Range.Min < Min)
+            {   
+                quantityAdditionalBins = Convert.ToInt64(Math.Ceiling((Min - data.Elements.First())/BinWidth));
+                double[] newBinCounts = new double[quantityAdditionalBins + BinCounts.Length];
+
+                for (Int64 i = BinCounts.Length + quantityAdditionalBins -1; i > (quantityAdditionalBins-1); i--)
+                {
+                    newBinCounts[i] = BinCounts[i - quantityAdditionalBins];
+                }
+                BinCounts = newBinCounts;
+                BinCounts[0] += 1;
+                double newMin = Min - quantityAdditionalBins * BinWidth;
+                double max = Max;
+                Min = newMin;
+                Range = IRangeFactory.Factory(newMin, max, true, true, true, false);
+            } else if (data.Range.Max > Max)
+            {
+                quantityAdditionalBins = Convert.ToInt64(Math.Ceiling((data.Elements.First() - Max+BinWidth) / BinWidth));
+                double[] newBinCounts = new double[quantityAdditionalBins + BinCounts.Length];
+                for (Int64 i = 0; i < BinCounts.Length; i++)
+                {
+                    newBinCounts[i] = BinCounts[i];
+                }
+                newBinCounts[BinCounts.Length + quantityAdditionalBins-1] += 1;
+                BinCounts = newBinCounts;
+                double newMax = (BinCounts.Length +1) * BinWidth;
+                Max = newMax;
+                double min = Min;
+                Range = IRangeFactory.Factory(min, newMax, true, true, true, false);
+            } else
+            {
+                Int64 newObsIndex = Convert.ToInt64(Math.Floor((data.Elements.First() - Min) / BinWidth));
+                if (data.Elements.First() == Max)
+                {
+                    quantityAdditionalBins = 1;
+                    double[] newBinCounts = new double[quantityAdditionalBins + BinCounts.Length];
+                    for (Int64 i = 0; i < BinCounts.Length; i++)
+                    {
+                        newBinCounts[i] = BinCounts[i];
+                    }
+                    BinCounts = newBinCounts;
+                    double newMax = (BinCounts.Length + 1) * BinWidth;
+                    Max = newMax;
+                }
+                BinCounts[newObsIndex] += 1;
+            }
+        }
+
+        public void AddObservationsToHistogram(IData data)
```

Comment:
I would love to see this as implemented with an array of double instead of an IData

Reply:
can be done at a later date with a refactor